components/RenderProposalContentLambda.tsx

```diff
@@ -214,18 +230,41 @@ export const contentToData = (
         addresses: undefined,
         entrypoints: undefined,
         params: JSON.stringify(
-          lambdaData[0].txs.map(({ to_, token_id, amount }) => ({
-            fa2_address:
-              token?.token.contract.address ?? lambda?.contractAddress,
-            name: token?.token.metadata.name,
-            token_id,
-            to: to_,
-            amount: !!token?.token.metadata.decimals
-              ? BigNumber(amount)
-                  .div(BigNumber(10).pow(token?.token.metadata.decimals ?? 0))
-                  .toString()
-              : amount.toString() + "*",
-          }))
+          lambdaData[0].txs.map(({ to_, token_id, amount }) => {
+            const token = walletTokens.find(
+              token =>
+                token.token.contract.address === lambda?.contractAddress &&
+                token.token.tokenId === token_id.toString()
+            );
+            const metadata = token?.token.metadata;
+            let imageUri = metadata?.thumbnailUri;
+
+            if (
+              imageUri === undefined &&
+              metadata !== undefined &&
+              "displayUri" in metadata &&
+              metadata.displayUri !== undefined
```

Comment:
`imageUri === undefined` -> `!imageUri`
`metadata !== undefined` -> `metadata` OR `!!metadata`
same for `displayUri`

Reply:
The result is different. in the case of `imageUri` is empty string, which is possible. your version work for the empty string. I think, in general, explicit to check the type is good thing unless the writer is very clear what's writing for. I'd say it's hard for writer to always keep an eye such things. I changed for this case.